openedx/core/djangoapps/dark_lang/views.py

```diff
@@ -2,164 +2,112 @@
         """
-        context = {
-            'disable_courseware_js': True,
-            'uses_pattern_library': True
+        return {
+            'uses_bootstrap': True,
         }
-        return render_to_response(self.template_name, context)
 
-    @method_decorator(login_required)
-    def post(self, request):
+    def standalone_page_title(self, request, fragment, **kwargs):
         """
-        Sets or clears the DarkLang depending on the incoming post data.
-
-        Arguments:
-            request (Request): The Django Request Object
-
-        Returns:
-            HttpResponse: View containing the form for setting the preview lang with the status
-                included in the context
+        Returns the page title for the standalone update page.
         """
-        return self.process_darklang_request(request)
+        return _('Preview Language Administration')
 
-    def process_darklang_request(self, request):
+    @method_decorator(login_required)
+    def get(self, request, *args, **kwargs):
         """
-        Proccess the request to Set or clear the DarkLang depending on the incoming request.
-
-        Arguments:
-            request (Request): The Django Request Object
+        Renders the fragment to control the preview language.
+        """
+        if not self._user_can_preview_languages(request.user):
+            raise Http404
+        return super(PreviewLanguageFragmentView, self).get(request, *args, **kwargs)
 
-        Returns:
-            HttpResponse: View containing the form for setting the preview lang with the status
-                included in the context
+    @method_decorator(login_required)
+    def post(self, request, **kwargs):
+        """
+        Accept requests to update the theme preview.
+        """
+        if not self._user_can_preview_languages(request.user):
+            raise Http404
+        action = request.POST.get('action', None)
+        if action == 'set_preview_language':
+            self._set_preview_language(request)
+        elif action == 'reset_preview_language':
+            self._clear_preview_language(request)
+        return redirect(request.path)
+
+    def _user_can_preview_languages(self, user):
+        """
+        Returns true if the specified user can preview languages.
         """
-        context = {
-            'disable_courseware_js': True,
-            'uses_pattern_library': True
-        }
-        response = None
         if not DarkLangConfig.current().enabled:
-            message = _('Preview Language is currently disabled')
-            context.update({'form_submit_message': message})
-            context.update({'success': False})
-            response = render_to_response(self.template_name, context, request=request)
-
-        elif 'set_language' in request.POST:
-            # Set the Preview Language
-            response = self._set_preview_language(request, context)
-        elif 'reset' in request.POST:
```

Comment:
Reset is working for me. If I enter the language code "eo" then the page switches, and then clicking reset it reverts back to English:

https://andy-armstrong.sandbox.edx.org/update_lang/

I didn't look into any existing tests. I would guess there aren't any.